src/lager_util.erl

```diff
@@ -459,22 +459,15 @@ i3l(I) when I < 100 -> [$0 | i2l(I)];
 i3l(I)              -> integer_to_list(I).
 
 %% When log_root option is provided, get the real path to a file
-expand_path(RelPath) ->
-    RelPath2 = case application:get_env(lager, log_root) of
-                   {ok, LogRoot} when is_list(LogRoot) -> % Join relative path
-                       %% check if the given RelPath contains LogRoot, if so, do not add
-                       %% it again; see gh #304
-                       case filename:dirname(RelPath) of
-                           "." ->
-                               filename:join(LogRoot, RelPath);
-                           false ->
-                               RelPath
-                       end;
-                   undefined -> % No log_root given, keep relative path
-                       RelPath
-               end,
-    %% see #534 make sure c:cd can't change file path, trans filename to abs name
-    filename:absname(RelPath2).
+expand_path(LogPath) ->
+    IsLogPathAbsolute = filename:absname(LogPath) =:= LogPath,
+    case IsLogPathAbsolute of
```

Comment:
Suggest removing this intermediate var and just doing the check in the case head?

Reply:
Sure.

Reply:
Squashed over the previous commit. I still have to look at why the tests are failing, though.